src/engine/QMLEngine.js

```diff
@@ -493,12 +498,23 @@ class QMLEngine {
 
   // Return a path to load the file
   $resolvePath(file) {
+    if (this.$rootBasePath === undefined || this.$basePath === undefined) {
+      return file;
+    }
+
+    // "Root" references in QML should still be relative to the root QML file
+    if (file.indexOf("/") === 0 && this.$rootBasePath !== undefined) {
```

Comment:
> Are you good with this concept of rootBasePath?

I think yes, we could improve things later if needed, and this looks like a clear improvement to me atm.

I am not sure if we should store `$rootBasePath` though, as we could just do `const rootBasePath = this.$basePath.slice(0, this.$basePath.lastIndexOf('/'))` directly whenever it's used, and `this.$rootBasePath === undefined` is equivalent to `this.$basePath === undefined` in the current code.

> It requires anything outside of wherever the root QML file is stored to have a full HTTP path.

Could you provide an example where this would break?